core/engine_test.go

```diff
@@ -968,3 +963,70 @@ func TestMinIterationDuration(t *testing.T) {
 	// But we expect the custom counter to be added to 4 times
 	assert.Equal(t, 4.0, getMetricSum(collector, "testcounter"))
 }
+
+//nolint: funlen
+func TestMinIterationDurationInSetupTeardownStage(t *testing.T) {
+	t.Parallel()
+	setupScript := `
+		import { sleep } from "k6";
+
+		export function setup() {
+			sleep(1);
+		}
+
+		export let options = {
+			minIterationDuration: "2s",
+			duration: "2s",
+			setupTimeout: "2s",
+		};
+
+		export default function () {
+		};`
+	teardownScript := `
+		import { sleep } from "k6";
+
+		export let options = {
+			minIterationDuration: "2s",
+			duration: "2s",
+			teardownTimeout: "2s",
+		};
+
+		export default function () {
+		};
+
+		export function teardown() {
+			sleep(1);
```

Comment:
Nit: could we lower the sleep and duration times in both tests? This should pass if you instead use milliseconds, but would make a noticeable difference for humans. :)

Reply:
I think we cant, sleep somehow does not support float

Reply:
[What do you mean?](https://github.com/loadimpact/k6/blob/master/samples/thresholds_readme_example.js#L62)

Reply:
@imiric I did tried using small time duration, but it doesn't work. You can check in commit history.